src/ssh/azext_ssh/custom.py

```diff
@@ -105,7 +115,10 @@ def _check_or_create_public_private_files(public_key_file, private_key_file):
         ssh_utils.create_ssh_keyfile(private_key_file)
 
     if not public_key_file:
-        raise util.CLIError(f"Public key file not specified")
+        if private_key_file:
+            public_key_file = private_key_file + ".pub"
+        else:
+            raise util.CLIError("Public key file not specified")
```

Comment:
Could you please use a specific error type instead of `CLIError`

Reply:
Are there docs or examples? The existing code through the ssh extension seems to raise only two error types.

Reply:
Please refer to this document to set specific error types [error_handling_guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)